profitbricks-rest/src/main/java/org/apache/jclouds/profitbricks/rest/features/ServerApi.java

```diff
@@ -0,0 +1,212 @@
+import org.apache.jclouds.profitbricks.rest.domain.options.DepthOptions;
+import org.jclouds.Fallbacks;
+import org.jclouds.Fallbacks.EmptyListOnNotFoundOr404;
+import org.jclouds.http.filters.BasicAuthentication;
+import org.jclouds.http.functions.ParseJson;
+import org.jclouds.json.Json;
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.MapBinder;
+import org.jclouds.rest.annotations.PATCH;
+import org.jclouds.rest.annotations.PayloadParam;
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.ResponseParser;
+import org.jclouds.rest.annotations.SelectJson;
+import org.jclouds.util.Strings2;
+
+@Path("/datacenters/{dataCenterId}/servers")
+@RequestFilters(BasicAuthentication.class)
+public interface ServerApi extends Closeable {
+   
+   @Named("server:list")
+   @GET
+   @SelectJson("items")
+   @Fallback(EmptyListOnNotFoundOr404.class)
+   List<Server> getList(@PathParam("dataCenterId") String dataCenterId);
+
+   @Named("server:list")
+   @GET
+   @SelectJson("items")
+   @Fallback(EmptyListOnNotFoundOr404.class)
+   List<Server> getList(@PathParam("dataCenterId") String dataCenterId, DepthOptions options);
+   
+   @Named("server:get")
+   @GET   
+   @Path("/{serverId}")
+   @ResponseParser(ServerApi.ServerParser.class)
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   Server getServer(@PathParam("dataCenterId") String dataCenterId, @PathParam("serverId") String serverId);
+
+   @Named("server:get")
+   @GET   
+   @Path("/{serverId}")
+   @ResponseParser(ServerApi.ServerParser.class)
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   Server getServer(@PathParam("dataCenterId") String dataCenterId, @PathParam("serverId") String serverId, DepthOptions options);
+   
+   @Named("server:create")
+   @POST
+   @MapBinder(CreateServerRequestBinder.class)
+   @ResponseParser(ServerApi.ServerParser.class)
+   Server createServer(@PayloadParam("server") Server.Request.CreatePayload payload);
+   
+   @Named("server:update")
+   @PATCH
+   @Path("/{serverId}")
+   @MapBinder(UpdateServerRequestBinder.class)
+   @ResponseParser(ServerApi.ServerParser.class)
+   @Produces("application/vnd.profitbricks.partial-properties+json")
+   Server updateServer(@PayloadParam("server") Server.Request.UpdatePayload payload);
+   
+   @Named("server:delete")
+   @DELETE
+   @Path("/{serverId}")
+   @Fallback(Fallbacks.VoidOnNotFoundOr404.class)
+   void deleteServer(@PathParam("dataCenterId") String dataCenterId, @PathParam("serverId") String serverId);
+   
+   @Named("server:volume:list")
+   @GET
+   @Path("/{serverId}/volumes")
+   @Fallback(EmptyListOnNotFoundOr404.class)
+   @SelectJson("items")
+   List<Volume> listAttachedVolumes(@PathParam("dataCenterId") String dataCenterId, @PathParam("serverId") String serverId);
+   
+   @Named("server:volume:attach")
+   @POST
+   @Path("/{serverId}/volumes")
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
```

Comment:
Remove 404 fallback for POST/PUT APIs. ([discussion](https://github.com/jclouds/jclouds-labs-openstack/pull/135/files#r16700137))